model/api/src/main/java/org/jboss/aerogear/unifiedpush/api/validation/DeviceTokenValidator.java

```diff
@@ -68,6 +68,18 @@ public boolean isValid(Installation installation, ConstraintValidatorContext con
         }
         final VariantType type = installation.getVariant().getType();
 
+        return isValidDeviceTokenForVariant(deviceToken, type);
+    }
+
+
+    /**
+     * Helper to run quick up-front validations.
+     *
+     * @param deviceToken the submitted device token
+     * @param type type of the variant
+     * @return true if the token is valid
+     */
+    public static boolean isValidDeviceTokenForVariant(final String deviceToken, final VariantType type) {
```

Comment:
Maybe for another PR but I think we should log when the token is invalid, now the only feedback , is the endpoint returning a bad request but it can be hard for one to find out the reason.